components/certificate-mgt/org.wso2.carbon.identity.certificate.management.core/src/main/java/org/wso2/carbon/identity/certificate/management/core/dao/impl/CertificateManagementDAOImpl.java

```diff
@@ -0,0 +1,192 @@
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.certificate.management.core.dao.impl;
+
+import org.wso2.carbon.database.utils.jdbc.NamedJdbcTemplate;
+import org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException;
+import org.wso2.carbon.database.utils.jdbc.exceptions.TransactionException;
+import org.wso2.carbon.identity.certificate.management.core.constant.CertificateMgtErrors;
+import org.wso2.carbon.identity.certificate.management.core.constant.CertificateMgtSQLConstants;
+import org.wso2.carbon.identity.certificate.management.core.dao.CertificateManagementDAO;
+import org.wso2.carbon.identity.certificate.management.core.exception.CertificateMgtException;
+import org.wso2.carbon.identity.certificate.management.core.model.Certificate;
+import org.wso2.carbon.identity.certificate.management.core.util.CertificateMgtUtil;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class implements the Certificate Management DAO.
+ * And this is responsible for handling the CRUD operations with the database.
+ */
+public class CertificateManagementDAOImpl implements CertificateManagementDAO {
+
+    @Override
+    public void addCertificate(String certificateId, Certificate certificate, int tenantId)
+            throws CertificateMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            InputStream certByteStream = CertificateMgtUtil.getCertificateByteStream(certificate.getCertificate());
+            int certificateLength = certByteStream.available();
+            jdbcTemplate.withTransaction(template -> {
+                template.executeInsert(CertificateMgtSQLConstants.Query.ADD_CERTIFICATE,
+                    preparedStatement -> {
+                        int index = 1;
+                        preparedStatement.setString(index++, certificateId);
+                        preparedStatement.setString(index++, certificate.getName());
+                        preparedStatement.setBlob(index++, certByteStream, certificateLength);
+                            preparedStatement.setInt(index, tenantId);
+                        }, certificate, false);
+                return null;
```

Comment:
why we have this null return ?

Reply:
The line return null; inside the transaction block is used because the withTransaction method expects a return value, but in this case, the code does not need to return anything meaningful.